operator/cmd/mesh/manifest-apply.go

```diff
@@ -95,6 +95,7 @@ func manifestApplyCmd(rootArgs *rootArgs, maArgs *manifestApplyArgs, logOpts *lo
 `,
 		Args: cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Println("! istioctl manifest apply has graduated. Use istioctl install instead")
```

Comment:
Instead of this, can you use the `softGraduatedCmd()` mechanism that we use elsewhere?  (We may need another version of softGraduatedCommand() because we rename "apply" to "install")  The idea behind softGraduatedCommand() and graduatedCommand() is that they also change the short description.

Reply:
I have tried writing deprecation command, but this requires [manifestApplyCmd](https://github.com/istio/istio/blob/master/operator/cmd/mesh/manifest-apply.go#L78) to be modified,exported and needs few other changes to work with following function. 
```go
// deprecatedCmd is used for commands that have deperacted and replaced by new command.
func deprecatedCmd(oldCmd, newCmd string) *cobra.Command {
	msg := fmt.Sprintf("(%s has deprecated. Use `istioctl %s`)", oldCmd, newCmd)
	return &cobra.Command{
		Use:   newCmd,
		Short: msg,
		RunE: func(_ *cobra.Command, _ []string) error {
			return errors.New(msg)
		},
	}
}
```
As mentioned by @ostromart in  https://github.com/istio/istio.io/pull/7396#pullrequestreview-421110570 `manifest apply` is going to be removed in 1.7, so we can go ahead and delete [code related to `manifest apply`](https://github.com/istio/istio/blob/master/operator/cmd/mesh/manifest-apply.go#L78).

I will prefer we keep this PR as it is without introducing too much code changes. The reason is that because this code will be cherry-picked in `1.6` and will be unused in `1.7` . We have already updated `1.6` doc to use `istioctl install` here https://github.com/istio/istio.io/pull/7396. 
 



@esnible Let me know if this sounds ok?

Reply:
@shamsher31 cherrypicks to 1.6 should be done with care and all work should be atomic in nature. If your proposal is to iterate, then this work should not be cherrypicked.

Reply:
@sdake, I agree, we should be more careful about what we are picking in 1.6. 

What I mean by iterating is that the deprecated flag `manifest apply` will be removed in 1.7 so we shouldn't be spending much time and energy on that. Keeping the deprecation message as simple as possible to avoid more issues in future releases.

Reply:
@esnible any thoughts on this?

Reply:
It is too late to remove `manifest apply` in 1.7.  We didn't deprecate it in 1.6.0.  We can deprecate it in 1.7 and remove it in 1.8.

Reply:
I'm unclear about the deprecation of `istioctl manifest apply`. Saw this [comment](https://github.com/istio/istio.io/pull/7502/files#diff-eb6e9d0c5d20f9923cc19b8b24fc8f14R44) . Also added by @ostromart https://github.com/istio/istio/pull/24509 which will show the message during `--help`. 

Not sure if we can close this PR or it will be good to show deprecation message during `manifest apply`

Reply:
Keeping it aligned with https://github.com/istio/istio/pull/24509